x-pack/legacy/plugins/apm/public/utils/flattenObject.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { compact, isObject, isEmpty } from 'lodash';
+
+export interface KeyValuePair {
+  key: string;
+  value: unknown;
+}
+
+export const flattenObject = (
+  item: Record<string, any>,
```

Comment:
Since this also accepts arrays shouldn't this be 
```suggestion
  item: Record<string, any> |  any[],
```

?

Reply:
Oh, and since we now allow it to be empty:
```suggestion
  item: Record<string, any> |  any[] | null | undefined,
```